### PR TITLE
Ethernet_Change_Documentation

### PR DESCRIPTION
**Pull request by Siddhant779**
I changed the documentation in the drivers.rst file.

### CODE REVIEW

File: Docs/source/Software/drivers.rst [hunk 1]
```diff
@@ -16,10 +16,28 @@ Ethernet
...
 Usage
 
+Ethernet_Init() is first called as a way to create the Queue and create a socket when a connection has been established.
+Ethernet_ConnectToServer() is a static blocking function called inside Ethernet_Init() that is either called when connection
+first needs to be established or when connection is lost with telemetry.  Ethernet_SendMessage() then collects data from the 
+Ethernet Queue and sends through Ethernet - if there is no connection, it calls Ethernet_ConnectToServer().Ethernet_SendMessage() returns pdFalse
+if the Queue is empty and pdTrue if the Queue is not empty. The struct EthernetMSG_t will hold the data packets of either information 
+regarding IMU, GPS, or CAN. The struct consists of an id enum, length, and a union of data. Since the data is in a union, there will be zero padding
+at the MSB.
+
```
Comment (manthanand) [inline Docs/source/Software/drivers.rst]: Markup all the function names. Also check to see if this compiles when you run 'make html'. Make sure everything you wrote looks the way you want it to when you run it.

File: Docs/source/Software/drivers.rst [hunk 2]
```diff
@@ -16,10 +16,28 @@ Ethernet
...
 Additional Considerations
 
+Sunlight could have been implemented as a server instead of a client. This implementation would make more sense if Ethernet had the option of sending
+messages to multiple clients. The way that it is implemented, Ethernet is trying to connect to telemetry so that it can send information. 
+As sunlight as the server, telemetry would try and connect to sunlight in order to send information. In order to implement these changes, both
```
Comment (manthanand) [inline Docs/source/Software/drivers.rst]: Sunlight and telemetry are the same thing.

**Review by Siddhant779 — COMMENTED** [review]
I also changed documentation

File: Docs/source/Software/drivers.rst [hunk 3]
```diff
@@ -11,20 +11,31 @@ Usage
...
 Usage
 
-
+``Ethernet_Init()`` is first called as a way to create the Queue and create a socket when a connection has been established.
+``Ethernet_ConnectToServer()`` is a static blocking function called inside ``Ethernet_Init()`` that is either called when connection
+first needs to be established or when connection is lost with data acquisition.  ``Ethernet_SendMessage()`` then collects data from the 
+Ethernet Queue and sends through Ethernet - if there is no connection, it calls ``Ethernet_ConnectToServer()``. ``Ethernet_SendMessage()`` returns pdFalse
+if the Queue is empty and pdTrue if the Queue is not empty. The struct EthernetMSG_t will hold the data packets of either information 
+regarding IMU, GPS, or CAN. The struct consists of an id enum, length, and a union of data. Since the data is in a union, there will be zero padding
+at the MSB.
```
Comment (manthanand) [inline Docs/source/Software/drivers.rst]: You don't actually talk about how to use the driver in this documentation. There isn't any mention of Ethernet_PutInQueue which is what actually allows the layers above to put messages in the queue. Most of these just discuss what the functions are, not how to use them.

**Review by manthanand — APPROVED** [review]

**Pull request merged**